sql/catalyst/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriterWrapper.scala

```diff
@@ -60,6 +60,13 @@ case class ArrowWriterWrapper(
       allocator = null
     }
   }
+
+  /**
+   * Clear the underlying VectorSchemaRoot to release the memory held by the vectors.
```

Comment:
what's the difference between `clear` and `close`? The wrier can be reused after cleared but not closed?

Reply:
`clear` and `close` are same on `FieldVector`. `VectorSchemaRoot.clear` additionally resets its `rowCount`, so yes, `VectorSchemaRoot` can be reused after `clear`.

```
/** Alternative to clear(). Allows use as an AutoCloseable in try-with-resources. */
@Override
void close();
```